scripts-dev/release.py

```diff
@@ -396,6 +385,13 @@ def upload():
     current_version = get_package_version()
     tag_name = f"v{current_version}"
 
+    # Check we have the right tag checked out.
+    repo = get_repo_and_check_clean_checkout()
+    tag = repo.tag(f"refs/tags/{tag_name}")
+    if repo.head.commit != tag.commit:
+        click.echo("Tag {tag_name} (tag.commit) is not currently checked out!")
+        click.get_current_context().abort()
```

Comment:
I don't think this is actually necessary for `upload` as we don't interact with the repo at all, we just download and reupload files.

Though might make sense to keep the check purely for sanity reasons.

Reply:
Oh, I didn't realise we downloaded the wheel and sdist; I assumed we'd build them locally.

I guess I'll leave it in. (The line echoes an instruction to do a git merge, so it can't hurt)